### DISCUSSION TITLE
Proxy to Graphql

### DISCUSSION THREAD

**Discussion by billmcknight1953** [chimurai/http-proxy-middleware#542] [Q&A]
I am having difficulty proxying to a graphql endpoint whose endpoint is   http://XXX.XXX.XXX.XXX:8080/graphql.   The http-proxy-middleware code I am using in my express.js host is straightforward and simple:

```
app.use('/graphql', proxy({
   target: 'http://XXX.XXX.XXX.XXX:8080',
   changeOrigin: true
}));

```
The problem is that the response to this call is Bad Request (400) if the content type is application/json, which is the default from my Apollo grapqhl client.   But curiously if I call the proxied endpoint using text/plain in POSTMAN, it works.  If I call the graphql endpoint itself directly with POSTMAN, application/json it works as well.  Anyone know why I am getting a Bad Request response?

**Comment by MatthiasGrandl** [comment]
This is probably caused by the `body-parser` middleware.

**Comment by ryudo2k5** [comment]
Also getting the same issue

**Reply by gepdev** [reply]
Someone found a solution to this? For me it works fine through postman but not on browser.

**Reply by chimurai** [reply]
Can you provide a minimal working reproduction? Without one it's a guessing game.

**Reply by gepdev** [reply]
> Can you provide a minimal working reproduction? Without one it's a guessing game.


This is what I have done. I have tried to change response but when intercepting it the body is the query payload when in browser or the actual response data when in postman. 
```
const express = require("express");
const { createProxyMiddleware } = require("http-proxy-middleware");
const cors = require("cors");
const app = express();
app.use(cors());

const proxy = createProxyMiddleware({
  target: "https://exampleurl",
  changeOrigin: true,
  ws: false,
  logLevel: "debug"
});

app.use("/graphql*", proxy);
app.listen(8012);
```

My dependencies:
`"express": "^4.18.1"`
`"http-proxy-middleware": "^2.0.6"`

**Reply by chimurai** [reply]
Did you try to remove the `cors()` middleware, to see if that is interfering with hpm?

Otherwise you could compare the browser's `curl` request with postman's `curl` request. 

- https://everything.curl.dev/usingcurl/copyas#from-chrome-and-edge
- https://codingnconcepts.com/postman/how-to-generate-curl-command-from-postman/#generate-curl-from-postman

**Reply by VityaSchel** [reply]
> Did you try to remove the `cors()` middleware, to see if that is interfering with hpm?
> 
> Otherwise you could compare the browser's `curl` request with postman's `curl` request.
> 
> * https://everything.curl.dev/usingcurl/copyas#from-chrome-and-edge
> * https://codingnconcepts.com/postman/how-to-generate-curl-command-from-postman/#generate-curl-from-postman

Without cors() middleware it's impossible to make a request to the proxy, it fails with CORS error in browser.
